Packs/Campaign/Scripts/PerformActionOnCampaignIncidents/PerformActionOnCampaignIncidents.py

```diff
@@ -1,202 +1,525 @@
-    demisto.debug(f"Added campaign {campaign_id} to incident {incident_id}")
+    Args:
+        incident_context (dict): The incident context.
 
+    Returns:
+        dict: A dictionary mapping email field names to their values.
+
+    Note:
+        If an email field is None, it is replaced with an empty list.
+    """
+    return {field: (_get_data_from_incident(incident_context, field) or []) for field in RECIPIENT_FIELDS}
 
-def _remove_incident_from_lower_similarity_context(incident_context, incident_ids):
-    lower_similarity_incident_context = demisto.dt(incident_context,
-                                                   'Contents.context.EmailCampaign.LowerSimilarityIncidents')
 
-    lower_similarity_incident_context = list(filter(
-        lambda x: x.get('id') not in incident_ids, lower_similarity_incident_context))
+def _get_recipients(email_address_data: dict) -> list:
+    """Extracts recipients from the email fields.
 
-    demisto.executeCommand('DeleteContext', {'key': 'EmailCampaign.LowerSimilarityIncidents'})
+    Args:
+        email_address_data (dict): A dictionary mapping email field names to their values.
 
-    res = demisto.executeCommand('SetByIncidentId', {'key': 'EmailCampaign.LowerSimilarityIncidents',
-                                                     'value': lower_similarity_incident_context})
-    if is_error(res):
-        return_error('Failed to change context. Error details:\n{}'.format(get_error(res)))
+    Returns:
+        list: A list of recipients.
+    """
+    list_recipients: list = []
+    for email_field in RECIPIENT_FIELDS:
+        for recipient in email_address_data[email_field]:
+            if recipient:
+                list_recipients.append(recipient)
+    return list_recipients
 
 
-def perform_add_to_campaign(ids, action):
-    demisto.debug('starting add to campaign')
-    campaign_id = demisto.incident()['id']
-    campaign_incident_context = demisto.executeCommand('getContext', {'id': campaign_id})
-    demisto.debug(f'got incident context: {campaign_incident_context}')
+def _extract_incident_fields(incident_context, recipients: list) -> dict:
+    """Extracts relevant fields from the incident context.
+
+    Args:
+        incident_context (dict): The incident context.
+        recipients (list): A list of recipients.
+
+    Returns:
+        dict: A dictionary mapping field names to their values.
+    """
+    email_from = list(always_iterable(_get_data_from_incident(incident_context, "CustomFields.emailfrom")))
+    email_from_domain = [extract_domain(email) for email in email_from]
+    recipients_domain = [extract_domain(email) for email in recipients]
+    return {
+        "similarity": None,
+        "occurred": _get_data_from_incident(incident_context, "occurred")[0],
+        "emailfrom": extract_single_or_list(email_from),
+        "emailfromdomain": extract_single_or_list(email_from_domain),
```

Comment:
Why dont just put a list here? even if its one address?

Reply:
As the incident that added by the playbook